### ISSUE TITLE
merge wolf code into arepl?

### ISSUE THREAD

**Issue by Almenon** [Almenon/AREPL-vscode#274]
I've also been thinking about talking to duroktar about merging arepl/wolf.  Arepl could stand to benefit with wolf's UI / inline style and in return I could rebrand arepl as wolf, vastly expanding wolf's potential user base. I think we could probably work out some sort of agreement that would be beneficial to us both.  But practically speaking merging the code might incur significant rework to adapt it such that it would be the same difficulty just to do it myself.

**Comment by Almenon** [comment]
Some Notes I took while looking into it:

do I really need entire backend? Maybe I could just copy frontend
^ i want to do the least work possible^^^

Wolf uses regex to parse string output of hunter. preferably I would use some other way to avoid regex based bugs.

benefits of using hunter?

or bdb?

or stick with exec()?

----------------------------------------------

format arepl output similar to output of WOLF <- see types.ts
then i could apply same decoration code (or just look at it as reference)

-----------------------------------------------

wolf vs arepl:
wolf uses hunter - different from bdb in that it offers a simple python API
with hunter you pass in a module and a action
ex: 

    import hunter
    hunter.trace(module='posixpath', action=hunter.CodePrinter)

    import os
    os.path.join('a', 'b')

this results in stdout trace
i can pass in my own functions to be executed

**Comment by Almenon** [comment]
I got the go-ahead from duoktar 👍

**Comment by goyalyashpal** [comment]
is this done with?

**Comment by Almenon** [comment]
@goyalyashpal No. Unlikely to be done in the near future, I'm not putting much work into this project currently.
